From c38b7d379bdfbd41897330b6d779d56d200d418a Mon Sep 17 00:00:00 2001
From: gorazdze <gorazd@dinamico.si>
Date: Tue, 29 May 2018 11:47:53 +0200
Subject: [PATCH] Add menu links alter

---
 src/Entity/TaxonomyMenu.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Entity/TaxonomyMenu.php b/src/Entity/TaxonomyMenu.php
index f9512a6..52fb372 100644
--- a/src/Entity/TaxonomyMenu.php
+++ b/src/Entity/TaxonomyMenu.php
@@ -202,6 +202,9 @@ class TaxonomyMenu extends ConfigEntityBase implements TaxonomyMenuInterface {
       $links[$mlid] = $this->buildMenuDefinition($term, $base_plugin_definition);
     }
 
+    // Alter menu links.
+    \Drupal::service('module_handler')->alter('taxonomy_menu_links', $links);
+
     return $links;
   }
 
-- 
2.17.0

