From f6aecf4ec8147f3a70250c4c394b01d45474bebf Mon Sep 17 00:00:00 2001
From: gorazdze <gorazd@dinamico.si>
Date: Tue, 12 Sep 2017 03:46:37 +0200
Subject: [PATCH] Added drupal_block_plugin function.

---
 src/TwigExtension.php | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/src/TwigExtension.php b/src/TwigExtension.php
index 676310f..b1bcdd9 100644
--- a/src/TwigExtension.php
+++ b/src/TwigExtension.php
@@ -24,6 +24,7 @@ class TwigExtension extends \Twig_Extension {
     return [
       new \Twig_SimpleFunction('drupal_view', 'views_embed_view'),
       new \Twig_SimpleFunction('drupal_block', [$this, 'drupalBlock']),
+      new \Twig_SimpleFunction('drupal_block_plugin', [$this, 'drupalBlockPlugin']),
       new \Twig_SimpleFunction('drupal_region', [$this, 'drupalRegion']),
       new \Twig_SimpleFunction('drupal_entity', [$this, 'drupalEntity']),
       new \Twig_SimpleFunction('drupal_field', [$this, 'drupalField']),
@@ -86,6 +87,37 @@ class TwigExtension extends \Twig_Extension {
   }
 
   /**
+   * Builds the render array for the provided block plugin.
+   *
+   * @param mixed $plugin_id
+   *   The ID of the block plugin to render.
+   * @param array $config
+   *   (Optional) Block configuration.
+   * @param bool $check_access
+   *   (Optional) Indicates that access check is required.
+   *
+   * @return null|array
+   *   A render array for the block or NULL if the block does not exist.
+   */
+  public function drupalBlockPlugin($plugin_id, $config = [], $check_access = TRUE) {
+    $block_plugin_manager = \Drupal::service('plugin.manager.block');
+    $block = $block_plugin_manager->createInstance($plugin_id, $config);
+
+    $access = TRUE;
+    if ($check_access) {
+      $access_result = $block->access(\Drupal::currentUser());
+      if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+        $access = FALSE;
+      }
+    }
+
+    if ($block && $access) {
+      $build = $block->build();
+      return $build;
+    }
+  }
+
+  /**
    * Builds the render array of a given region.
    *
    * @param string $region
-- 
2.10.1

